errors.go

```diff
@@ -21,6 +23,10 @@ func httpError(w http.ResponseWriter, r *http.Request, errorCode int, errorMessa
 	}).Errorf(errorMessage, args...)
 	msg := fmt.Sprintf(errorMessage, args...)
 	msg += "\r\nrequest-id: " + rid
+	if r.Body != nil {
+		io.Copy(ioutil.Discard, r.Body)
+		r.Body.Close()
```

Comment:
also, clever to put this in the error handler. I guess go sync storage does this too, but for some reason I thought we'd end up having to do this all over the place. :1st_place_medal: